### PR TITLE
CB-1353. Need trial license for Management Service

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

CM/CDH cluster creation fails in certain test environments due to using test license.  The change here is to start the trial period if Management Service is being setup.  The logic is extracted from LDAP setup service.

## How was this patch tested?

Tested actual cluster deployment.  Also tested that duplicate "begin trial" requests are handled (to account for multiple independent services that require the trial).

### CODE REVIEW

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLdapService.java [hunk 1]
```diff
@@ -35,29 +33,20 @@
...
-                umsClient.getAccountDetails(userCrn, userCrn, Optional.empty()).getClouderaManagerLicenseKey())) {
-            LOGGER.info("Enabling trial license.");
-            clouderaManagerResourceApi.beginTrial();
-        } else {
-            LOGGER.info("UMS detected and license key available, skipping trial license.");
-        }
 
         if (ldapConfig != null) {
+            ApiClient client = clouderaManagerClientFactory.getClient(stack, cluster, clientConfig);
+            licenseService.beginTrialIfNeeded(stack.getCreator(), client);
```
Comment (meowfaceman) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLdapService.java]: Now that we have a license service (good work), we should really pull the license service out of the LdapService -- IMO, it really doesn't belong here.

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLicenseService.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+
+    private void beginTrial(ApiClient apiClient) throws ApiException {
+        LOGGER.info("Enabling trial license.");
+
+        try {
+            clouderaManagerClientFactory.getClouderaManagerResourceApi(apiClient).beginTrial();
+        } catch (ApiException e) {
+            if (trialAlreadyStarted(e)) {
```
Comment (meowfaceman) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLicenseService.java]: Is this catch/continue block basically the real core difference between the previous behavior and the current behavior? Is the implication here is that this is getting run multiple times?

Comment (adoroszlai) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLicenseService.java]: The core difference is that trial is started (if needed) before adding MGMT service.  The catch/continue is there to allow both LDAP setup and MGMT service setup to do it independently (since I haven't find a way to query the trial state).  It may also help in case the task is retried -- don't know if that's a real scenario or not.

Comment (keyki) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLicenseService.java]: It is, because the flow steps are restartable.
